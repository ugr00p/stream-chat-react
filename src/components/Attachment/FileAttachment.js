import React from 'react';
import { FileIcon } from 'react-file-utils';
import prettybytes from 'pretty-bytes';
import { SafeAnchor } from '../SafeAnchor';

/** @type React.FC<import('types').FileAttachmentProps> */
const FileAttachment = ({ attachment }) => {
  return (
    <div
      data-testid="attachment-file"
      className="str-chat__message-attachment-file--item"
    >
      <FileIcon
        mimeType={attachment.mime_type}
        filename={attachment.title}
        big={true}
        size={30}
      />
      <div className="str-chat__message-attachment-file--item-text">
        <SafeAnchor href={attachment.asset_url} target="_blank" download>
          {attachment.title}
        </SafeAnchor>
        {attachment.file_size && (
          <span>{prettybytes(attachment.file_size)}</span>
        )}
      </div>
    </div>
  );
};

export default React.memo(FileAttachment);
