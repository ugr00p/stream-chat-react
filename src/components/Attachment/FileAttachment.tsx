import React from 'react';
import prettybytes from 'pretty-bytes';
import { FileIcon } from 'react-file-utils';

import { SafeAnchor } from '../SafeAnchor';

import type { Attachment } from 'stream-chat';

import type { DefaultAttachmentType } from '../../types/types';

export type FileAttachmentProps<At extends DefaultAttachmentType = DefaultAttachmentType> = {
  attachment: Attachment<At>;
};

const UnMemoizedFileAttachment = <At extends DefaultAttachmentType = DefaultAttachmentType>(
  props: FileAttachmentProps<At>,
) => {
  const { attachment } = props;

  return (
    <div className='str-chat__message-attachment-file--item' data-testid='attachment-file'>
      <FileIcon big={true} filename={attachment.title} mimeType={attachment.mime_type} size={30} />
      <div className='str-chat__message-attachment-file--item-text'>
        <SafeAnchor download href={attachment.asset_url} target='_blank'>
          {attachment.title}
        </SafeAnchor>
        {attachment.file_size && Number.isFinite(Number(attachment.file_size)) && (
          <span>{prettybytes(attachment.file_size)}</span>
        )}
      </div>
    </div>
  );
};

export const FileAttachment = React.memo(
  UnMemoizedFileAttachment,
) as typeof UnMemoizedFileAttachment;
