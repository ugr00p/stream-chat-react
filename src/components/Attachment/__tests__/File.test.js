import React from 'react';
import '@testing-library/jest-dom';
import renderer from 'react-test-renderer';
import FileAttachment from '../FileAttachment';

const getComponent = (attachment) => {
  return <FileAttachment attachment={attachment} />;
};

describe('File', () => {
  it('should render File component', () => {
    const file = {
      type: 'file',
      mime_type: 'application/pdf',
      title: 'Nice file',
      asset_url:
        'https://www.w3.org/WAI/ER/tests/xhtml/testfiles/resources/pdf/dummy.pdf',
      file_size: 1337,
      text: 'My file',
    };
    const tree = renderer.create(getComponent(file)).toJSON();
    expect(tree).toMatchSnapshot();
  });
});
