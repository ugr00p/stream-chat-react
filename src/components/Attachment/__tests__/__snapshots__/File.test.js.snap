// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`File should render File component 1`] = `
<div
  className="str-chat__message-attachment-file--item"
  data-testid="attachment-file"
>
  <svg
    height={30}
    style={
      Object {
        "maxWidth": "100%",
      }
    }
    viewBox="0 0 48 48"
    width={30}
  >
    <defs>
      <clipPath
        id="pageRadius1"
      >
        <rect
          height={48}
          rx={4}
          ry={4}
          width={40}
          x={4}
          y="0"
        />
      </clipPath>
      <clipPath
        id="foldCrop"
      >
        <rect
          height={12}
          transform="rotate(-45 0 12)"
          width={40}
        />
      </clipPath>
      <linearGradient
        id="pageGradient1"
        x1="100%"
        y1="0%"
        y2="100%"
      >
        <stop
          offset="0%"
          stopColor="white"
          stopOpacity={0.25}
        />
        <stop
          offset="66.67%"
          stopColor="white"
          stopOpacity="0"
        />
      </linearGradient>
    </defs>
    <g
      clipPath="url(#pageRadius1)"
      id="file"
    >
      <path
        d="M4 0 h 28 L 44 12 v 36 H 4 Z"
        fill="whitesmoke"
      />
      <path
        d="M4 0 h 28 L 44 12 v 36 H 4 Z"
        fill="url(#pageGradient1)"
      />
    </g>
    <g
      transform="translate(32 12) rotate(-90)"
    >
      <rect
        clipPath="url(#foldCrop)"
        fill="#dbdbdb"
        height={48}
        rx={4}
        ry={4}
        width={40}
      />
    </g>
    
  </svg>
  <div
    className="str-chat__message-attachment-file--item-text"
  >
    <a
      href="https://www.w3.org/WAI/ER/tests/xhtml/testfiles/resources/pdf/dummy.pdf"
      target="_blank"
    >
      Nice file
    </a>
    <span>
      1.34 kB
    </span>
  </div>
</div>
`;
